### DISCUSSION TITLE
How to add to TextLog without adding a newline

### DISCUSSION THREAD

**Discussion by jspv** [Textualize/textual#2965] [Q&A] [answered]
I'm trying to add text to TextLog without an automatic newline after every .write(), I can't seem to find a way to to this.  Any thoughts on the best approach?

**Comment by TomJGooding** [comment]
It looks like the newlines are baked into the TextLog, could you give an example of the sort of text you're wanting to write?

**Reply by jspv** [reply]
I’m outputting content being streamed in asynchronously from an LLM - it arrives one word at a time. 

Maybe a VerticalScroll container with a Static or Label?  But that does mean regenerating the entire area for each word.

**Reply by willmcgugan** [reply, marked as answer]
ChatGPT apps with Textual tend to use a widget with Markdown, and simply re-render the widget each time. It's not super efficient, but it works well enough.

I don't think we have a trivial way of updating just the last line of a TextLog, but it might be an idea for a future update.

**Comment by jspv** [comment]
I have this working, but came across a [bug](https://github.com/Textualize/textual/issues/3006) trying to render partially formatted Markdown tables.
